<h1><b>Analytics 512: Homework # 1</b></h1>

<h4><p><i>Arif Ali</i></p> 
<p>January 28th, 2016</p></h4>

<h2>Exercise 2.4 #2</h2>
<h3>Part A</h3>
<p>This is a regression problem because the objective is to figure out how the different explanatory variables affect CEO salary. Salary is not binned into categories, but rather is a number.</p> 
<p><i>n</i> is the 500 firms since they are the observations and <i>p</i> is profit, number of employees, and industry.</p>
<h3>Part B</h3>
<p>Since the response variable is binary, this is a classification problem. </p>
<p><i>n</i> is the observation of the 20 previous products and <i>p</i> is price charged for the product, marketing budget, competition price, and ten other variables (not the success/failure variable).</p>
<h3>Part C</h3>
<p>This is a regression problem because the percent change is trying to determine to exact change in percentage.</p>
<p><i>n</i> is the number of weeks during 2012 and <i>p</i> the % change in the dollar, the % change in the British market, and the % change in the German market.</p> 

<h2>Exercise 2.4 #4</h2>

<h3>Part A</h3>
<ol>
    <li>Determining whether a person has cancer or not. The Response would be a binary response of whether someone does or doesn't have cancer. This could be expanded to checking the specific type of cancer (ie lung, breast, skin, brain, etc). The predictors could be health measurements of an individual's vitals. An individual cancer history rating, etc. Goal is prediction; however, changes in vitals could be used as a measure of inference.</li>
    <li>Predicting what elected official wins. The Response is a candidate. Predictors could be money raises, whether a candidate is an incumbent, and the party the candidate is affiliated. County Statistics like average income or Census Data could be used. While the goal would be to predict, inference could be used to study with counties, a party should invest in campaigning in.</li>
    <li>Determining the reason behind whether a TV Series is cancel or renewed. This could be based on nielsen ratings, DVD sales, the cost of production. This seems more of an inference goal, trying to figure out what makes a show worth renewing.</li>
</ol>

<h3>Part B</h3>
<ol>
    <li>Determining the amount a indivdual makes. The response would be the income amount of a person. The Predictor would be credit history, payement, debt, assets. Goal would probably be more along the lines of inference because observing changes in the predictors might lead into insights as how much money an indivdual makes.</li>
    <li>Determing the amount of rain fall in a region. The response would the amount of rain and predictors could be time, amount of overcast, precipitation. The goal is prediction; the reason inference is probably not needed is because the relationships have probably already been uncovered.</li>
    <li>Determining the change in stock market. The Response would the amount the market increases or decreases. The Predictors could be currency rate changes, consumer sentiment, etc. This is a predicton goal.</li>
</ol> 

<h3>Part C</h3>
<ol>
    <li>Using cluster analysis to looking at build of blood cells could be used to identitify blood clothes.</li>
    <li>Cluster analysis could be used to find vehicle built-ups which could be useful in planning highways and extra roadlanes in certain areas.</li>
    <li>In a war zone, population clusters could be used by peacekeeping forcing to set up safe zones for refugees. Using cluster analysis could effectively identitify clusters.</li>
</ol> 

<h2>Exercise 2.4 #7</h2>

<h3>Part A</h3>

<table>
    <tr>
        <td>Observation</td>
        <td>Euclidean Distance</td>
    </tr>
    <tr>
        <td>1</td>
        <td>$$3$$</td>
    </tr>
    <tr>
        <td>2</td>
        <td>$$2$$</td>
    </tr>
    <tr>
        <td>3</td>
        <td>$$\sqrt{10}$$</td>
    </tr>
    <tr>
        <td>4</td>
        <td>$$\sqrt{5}$$</td>
    </tr>
    <tr>
        <td>5</td>
        <td>$$\sqrt{2}$$</td>
    </tr>
    <tr>
        <td>6</td>
        <td>$$\sqrt{3}$$</td>
    </tr>
</table>    

<h3>Part B</h3>
<p>Green, because observation 5 is the the $1^{st}$ nearest neighbor and the response for 5 is green.</p> 

<h3>Part C</h3>
<p>Red, 2 out of the 3 closest neighbors has a response of red.</p>

<h3>Part D</h3>
<p>Small, it would be more flexible whereas a bigger value of K would result in trying to accomodate more points.</p>

<h2>Exercise 2.4 #9</h2>

<h3>Part A</h3>

In [2]:
library("ISLR")
data(Auto)
Auto = na.omit(Auto)
Auto = Auto[!is.null(Auto),]
summary(Auto)


      mpg          cylinders      displacement     horsepower        weight    
 Min.   : 9.00   Min.   :3.000   Min.   : 68.0   Min.   : 46.0   Min.   :1613  
 1st Qu.:17.00   1st Qu.:4.000   1st Qu.:105.0   1st Qu.: 75.0   1st Qu.:2225  
 Median :22.75   Median :4.000   Median :151.0   Median : 93.5   Median :2804  
 Mean   :23.45   Mean   :5.472   Mean   :194.4   Mean   :104.5   Mean   :2978  
 3rd Qu.:29.00   3rd Qu.:8.000   3rd Qu.:275.8   3rd Qu.:126.0   3rd Qu.:3615  
 Max.   :46.60   Max.   :8.000   Max.   :455.0   Max.   :230.0   Max.   :5140  
                                                                               
  acceleration        year           origin                      name    
 Min.   : 8.00   Min.   :70.00   Min.   :1.000   amc matador       :  5  
 1st Qu.:13.78   1st Qu.:73.00   1st Qu.:1.000   ford pinto        :  5  
 Median :15.50   Median :76.00   Median :1.000   toyota corolla    :  5  
 Mean   :15.54   Mean   :75.98   Mean   :1.577   amc gremlin    

The quantitative predictors: mpg, displacement, horsepower, weight, and acceleration. 

The qualitative predictors: cylinders, year, origin, and name.

<h3>Part B</h3>

In [3]:
quantitative = c("mpg", "displacement", "horsepower", "weight", "acceleration")

partb = mapply(range, Auto[, quantitative])
rownames(partb) = c("lower", "upper")
partb

mpg displacement horsepower weight acceleration
lower  9.0           68         46   1613          8.0
upper 46.6          455        230   5140         24.8

### Part C

In [4]:
partc = cbind(mapply(mean, Auto[, quantitative]),mapply(sd, Auto[, quantitative]))
colnames(partc) = c("mean", "sd")
partc

mean         sd
mpg            23.44592   7.805007
displacement  194.41199 104.644004
horsepower    104.46939  38.491160
weight       2977.58418 849.402560
acceleration   15.54133   2.758864

### Part D

In [5]:
Auto_minus10_to_85 = Auto[,-(10:85)]
partd = cbind(t(mapply(range, Auto_minus10_to_85[, quantitative])), 
      mapply(mean, Auto_minus10_to_85[, quantitative]),
      mapply(sd, Auto_minus10_to_85[, quantitative]))
colnames(partd) = c("lower range","upper range", "mean", "sd")
partd

lower range upper range       mean         sd
mpg                    9        46.6   23.44592   7.805007
displacement          68       455.0  194.41199 104.644004
horsepower            46       230.0  104.46939  38.491160
weight              1613      5140.0 2977.58418 849.402560
acceleration           8        24.8   15.54133   2.758864

### Part E

In [6]:
png("HW1E9e.png")
pairs(Auto[,quantitative])
dev.off()

pdf 
  2

<p>For mpg, there seems to be a negative relation between it compared to displacement, horespower, and weight. The relation betwwen acceleration and mpg is not as well defined but seemingly more positive.</p>
<p>The is a positive relation between displacement compared to horsepower and weight. There seems to be a negative relationship between displacement and acceleration.</p>
<p>Horsepower and weight has a postive relationship and horsepower and acceleration have a negative relationship.</p>
<p>Relation weight and acceleration is less defined, but there seems to be a slightly negative relationship</p>

### Part F

The relation between mpg and horsepower seems to fit a line resembling $y=e^{-x}$. Displacement follows this line as well; however, the points in the middle are more spread out compared to mpg vs. horsepower.

## Exercise 2.4 #10

### Part A

In [7]:
library(MASS)

In [8]:
dim(Boston)
names(Boston)

[1] 506  14

[1] "crim"    "zn"      "indus"   "chas"    "nox"     "rm"      "age"    
 [8] "dis"     "rad"     "tax"     "ptratio" "black"   "lstat"   "medv"

<p>The rows represents observation in Bostonian suburbs; there are 506 observations, which seem to correspond to each Boston suburb. The columns are different attributes for the suburbs; there are 14 columns.

### Part B

In [9]:
png("HW1E10b.png")
pairs(Boston)
dev.off()

pdf 
  2

<p>Chas is obviously a binary variable based on how the scatter plots look. RM and medv seem to have a postive relationship; however, RM and lstat seems to have a negative relationship. Nox and dis have a relationship best describe by $y=e^{-x}$.</p> 

### Part C

In [10]:
png("HW1E10c.png")
par(mfrow = c(4,4))
for(i in names(Boston)){
    plot(Boston$crim, Boston[, i], xlab="crim", ylab = i)
    }
dev.off()

pdf 
  2

<p>It seems that the higher the age, the more crime. It could be that the older houses could be in more rundown neighborhoods; therefore, crime is more rampant. The lower the distance variable, the more crime, which is interesting because it seems counter-intuitive. In areas with higher tax rates, there is more crime. This could be because lower economic classes are taxed higher, and there is a relationship between poverty and crimes.</p>

### Part D

In [11]:
quantile(Boston$crim, seq(from = 0, to = 1, by = .1))
png("HW1E10di.png")
hist(Boston$crim, breaks = 25)
dev.off()

0%       10%       20%       30%       40%       50%       60%       70% 
 0.006320  0.038195  0.064170  0.099245  0.150380  0.256510  0.550070  1.728440 
      80%       90%      100% 
 5.581070 10.753000 88.976200

pdf 
  2

<p>The crime rate is pretty low throughout boston, with less than 40% of the suburb having a crime rate of greater than one. Based on the quantile, there seems to be a right tail as indicated. Boston seems to be a pretty safe place. However, there appears to be one suburb that has an abnormally high crime rate.</p> 

In [12]:
quantile(Boston$tax)
png("HW1E10dii.png")
hist(Boston$tax, breaks = 25)
dev.off()

0%  25%  50%  75% 100% 
 187  279  330  666  711

pdf 
  2

<p>Two of the suburbs have very high Tax rates. There is a significant gasp, which could indicate income inequality.</p>

In [13]:
quantile(Boston$ptratio)
png("HW1E10diii.png")
hist(Boston$ptratio, breaks = 10)
dev.off()

0%   25%   50%   75%  100% 
12.60 17.40 19.05 20.20 22.00

pdf 
  2

<p>There are an unusual number of suburbs with a ratio between 20 and 21. However no indication that there is any skew. Therefore none of the suburbs seem to have a particularily high pupil-teacher rate.</p>

### Part E

In [14]:
sum(Boston$chas == 1)

[1] 35

### Part F

In [15]:
median(Boston$ptratio)

[1] 19.05

### Part G

In [22]:
Boston[Boston$medv == min(Boston$medv),]

t(mapply(function(x) round(mean(x), digits = 4), Boston))

crim zn indus chas   nox    rm age    dis rad tax ptratio  black lstat
399 38.3518  0  18.1    0 0.693 5.453 100 1.4896  24 666    20.2 396.90 30.59
406 67.9208  0  18.1    0 0.693 5.683 100 1.4254  24 666    20.2 384.97 22.98
    medv
399    5
406    5

crim      zn   indus   chas    nox     rm     age   dis    rad      tax
[1,] 3.6135 11.3636 11.1368 0.0692 0.5547 6.2846 68.5749 3.795 9.5494 408.2372
     ptratio   black   lstat    medv
[1,] 18.4555 356.674 12.6531 22.5328

There are two suburbs with the lowest median house value referred to as min suburbs. In terms of crime, the min suburbs have significiantly higher rates. There are no zoned lots. Both have the same proportion of non-retail business acres per town, which was higher than the average. Neither min suburb borders the Charles River. There less average number of rooms in the min suburbs than overall. It looks like both suburbs have 100% of owner-occupied units built prior to 1940. The tax rate is higher than the average. The min suburbs are more accessible from the radial highways than the average suburbs. There is a slightly higher proportion of blacks in the min suburbs. The pupil teacher ratio is slightly higher in the min suburbs than overall. The weighted mean of distances to five Boston employment centres is shorter for the min suburbs.   

### Part H

In [17]:
sum(Boston$rm>7)
sum(Boston$rm>8)

[1] 64

[1] 13

In [38]:
t(mapply(function(x) round(mean(x), digits = 4),Boston[Boston$rm>8,]))


crim      zn  indus   chas    nox     rm     age    dis    rad      tax
[1,] 0.7188 13.6154 7.0785 0.1538 0.5392 8.3485 71.5385 3.4302 7.4615 325.0769
     ptratio    black lstat medv
[1,] 16.3615 385.2108  4.31 44.2

In [39]:
t(mapply(function(x) round(mean(x), digits = 4),Boston))

crim      zn   indus   chas    nox     rm     age   dis    rad      tax
[1,] 3.6135 11.3636 11.1368 0.0692 0.5547 6.2846 68.5749 3.795 9.5494 408.2372
     ptratio   black   lstat    medv
[1,] 18.4555 356.674 12.6531 22.5328

<p></p>

The suburbs where $rm > 8$ has a lower crime rate compared to the rest of the city. These suburbs are on average further than the average of the suburbs. Slight higher proportion are bordering the Charles River compared to the rest of the city. The Average age and median house value are higher. The tax rate is lower.